### DISCUSSION TITLE
Vue StoreFront Widgets

### DISCUSSION THREAD

**Discussion by ed42311** [algolia/instantsearch#5380] [Ideas]
**Feature ⚡️**

Hello, I'm building and integration with Algolia, Spryker and Vue Storefront.  Currently using the next branch of the vue storefront repo and I have setup the router from the docs to interact with algolia on a single page.

Vue storefront has a [library of UI components](https://github.com/vuestorefront/storefront-ui) which we are using to match theming, but I'm running into some issues with not being able to access data at depth, thus leading to building sort of hybrids of the Vue instant search components and the storefront UI components.

After running into this:  https://www.algolia.com/doc/guides/building-search-ui/widgets/create-your-own-widgets/vue/#existing-connectors

I was thinking that maybe I was going about it wrong, much of the problem was scoping and state...  what I'm wondering is... if I am encountering this problem, maybe this is an issue that is encountered by others.

### What is your use case for such a feature?

VSF integration with Algolia search

### What is your proposal

A prototype of one or two storefront widgets with connector mix-ins

### What is the version you are using?

    "vue-instantsearch": "^3.6.0"

> Always try the latest one before opening an issue.

**Comment by Haroenv** [comment]
Hi @ed42311, this is an interesting idea. Can you clarify more what you'd expect at the end of this? Are any new components needed, or mostly reskins of the existing ones? Thanks

**Comment by ed42311** [comment]
Don't know really yet.  At this point I have a single issue I'm trying to solve, which actually encapsulates the idea fairly well. 

I think... if am understanding the problem space correctly, that the end result would be a library or library subset that gives scope access to the state and result objects coming back from algolia within the organism and molecule level of the component space in storefront UI.

This could be paired with a the current custom router implementation to provide SSR state inside the ssr search wrapper.

Vue storefront uses these sets of composables, which I'm only tangentially familiar with at the moment (they look analogous to react hooks) so another solution might be to integrate directly into those...

 There is a possibility that I am missing something simple at this point that would make this conversation irrelevant

I have a hypothesis but I haven't tested it yet.  Maybe it would be good for me to build a repo with the problem to be solved in one branch and the hypothetical solution in another. I will post that when I have it.

**Comment by Haroenv** [comment]
That sounds very helpful, thanks for looking into this!
